shared/middleware/files_api.rb

```diff
@@ -141,20 +141,7 @@ def record_event(quota_event_type, quota_type, encrypted_channel_id)
     result[:body]
   end
 
-  #
-  # PUT /v3/(assets|sources)/<channel-id>/<filename>?version=<version-id>
-  #
-  # Create or replace a file. Optionally overwrite a specific version.
-  #
-  put %r{/v3/(assets|sources)/([^/]+)/([^/]+)$} do |endpoint, encrypted_channel_id, filename|
-    dont_cache
-
-    # read the entire request before considering rejecting it, otherwise varnish
-    # may return a 503 instead of whatever status code we specify. Unfortunately
-    # this prevents us from rejecting large files based on the Content-Length
-    # header.
-    body = request.body.read
-
+  def put_file(endpoint, encrypted_channel_id, filename, body)
```

Comment:
this makes me realize that our quota logic doesn't work quite right when we are replacing an existing file. Not sure if this is worth tracking or not.